### DISCUSSION TITLE
How to remove a queried entity in a system?

### DISCUSSION THREAD

**Discussion by Cympol** [SanderMertens/flecs#917] [Q&A]
*Using the C api.
In my game, entities can have a lifetime component, and when that count reaches zero, the entity is to be removed from the storage system. How do i get the entity id stored (if it is) in the ecs_iter_t value that a query returns, so i can remove them? Something like:
```
void SomeSystem(ecs_iter_t *it) {
    Lifetime *lifetime = ecs_field(it, Lifetime, 1);
    *lifetime -= it->delta_time;
    if (*lifetime < 0) {
--->    ecs_delete(it->world, it->"entity_I_want_to_remove"); 
    }
}
```

Edit, I think I figured it out. The field i was searching for was the "entities" one, and the code seems to work.

**Comment by SanderMertens** [comment]
Yup that's right, you would do:

```c
for (int i = 0; i < it->count; i ++) {
  ecs_delete(it->world, it->entities[i]);
}
```
